packages/vite/src/node/plugins/resolve.ts

```diff
@@ -488,32 +478,27 @@ function tryFsResolve(
   targetWeb = true,
   skipPackageJson = false,
 ): string | undefined {
-  let postfixIndex = fsPath.indexOf('?')
-  if (postfixIndex < 0) {
-    postfixIndex = fsPath.indexOf('#')
-
-    // Dependencies like es5-ext use `#` in their paths. We don't support `#` in user
-    // source code so we only need to perform the check for dependencies.
-    // We don't support `?` in node_modules paths, so we only need to check in this branch.
-    if (postfixIndex >= 0 && isInNodeModules(fsPath)) {
+  // Dependencies like es5-ext use `#` in their paths. We don't support `#` in user
+  // source code so we only need to perform the check for dependencies.
+  // We don't support `?` in node_modules paths, so we only need to check in this branch.
+  const hashIndex = fsPath.indexOf('#')
+  if (hashIndex > 0 && isInNodeModules(fsPath)) {
```

Comment:
```suggestion
  if (hashIndex >= 0 && isInNodeModules(fsPath)) {
```

Probably doesn't change anything, but maybe it should be `>=` as I think it's how it should work logically.